hs-abci-sdk/src/Tendermint/SDK/Application/Handlers.hs

```diff
@@ -165,18 +171,44 @@ makeHandlers (HandlersContext{..} :: HandlersContext alg ms core) =
         return . ResponseCommit $ def
           & Resp._commitData .~ Base64.fromBytes rootHash
 
+      beginBlock :: Handler 'MTBeginBlock (BA.BaseAppEffs core)
+      beginBlock (RequestBeginBlock bb) = do
+        catch
+          (do
+            res <- evalBeginBlockHandler $ M.applicationBeginBlocker app bb
+            return . ResponseBeginBlock $ res
+          )
+          (\(_ :: BA.AppError) -> return . ResponseBeginBlock $ def
```

Comment:
I'm kind of wondering if this is a place we should be using the [Exception](https://github.com/f-o-a-m/kepler/blob/master/hs-abci-types/src/Network/ABCI/Types/Messages/Response.hs#L527), which can be returned for any ABCI message which causes an internal error. I mean I would assume that if something fails at this point, it represents the fact that the chain is in an invalid state. Do you know if that's fair?

As it currently stands, the `catch` error handler will never actually be used, see the comment below about the `cache == Nothing` issue

Reply:
I tried to think of a reason a developer might want this to fail open and I really couldn't. Either this is a global failure and all validators will hit it, in which case failing open would keep consensus running but with possibly unexpected state. Or in the alternative, only some validators would error, and then they would halt at Commit when the state inconsistency was noticed. So I agree with making this an Exception, since then the cause is more obvious.